In [33]:
import pandas as pd
import numpy as np

midi_stats = pd.read_csv('midi_stats_v3_drop_cols_.csv',sep=",")
midi_stats

Unnamed: 0 first_time_signature  cant_time_signatures    bpm  compases  \
0              0                  3/4                     1   80.0        64   
1              1                  3/4                     1  100.0        64   
2              2                  3/4                     1   80.0        65   
3              3                  3/4                     1   90.0        80   
4              4                  3/4                     1   70.0        64   
...          ...                  ...                   ...    ...       ...   
4828        4828                  4/4                     1  100.0        34   
4829        4829                  4/4                     1  100.0        40   
4830        4830                  4/4                     1  100.0        42   
4831        4831                  4/4                     1  120.0       101   
4832        4832                  2/4                     1   80.0        95   

      cant_dist_chords  avg_notes_quantified  tempo_changes tonalidad  \
0                    0              0.038168              1         F   
1                    1              0.018692              1         G   
2                    6              0.000000              1         D   
3                    0              0.012862              1         G   
4                   10              0.036750              0         F   
...                ...                   ...            ...       ...   
4828                28              0.937018              0         E   
4829                21              0.878957              0         A   
4830                33              0.920635              0         A   
4831                96              0.363162              0         C   
4832                26              0.977612              0         C   

     tonalidad_escala  ...  ('Distortion Guitar', 0.25)  ('Tom', 0.25)  \
0                   M  ...                          NaN            NaN   
1                   M  ...                          NaN            NaN   
2                   m  ...                          NaN            NaN   
3                   M  ...                          NaN            NaN   
4                   M  ...                          NaN            NaN   
...               ...  ...                          ...            ...   
4828                m  ...                          NaN            NaN   
4829                m  ...                          NaN            NaN   
4830                m  ...                          NaN            NaN   
4831                m  ...                          NaN       0.524752   
4832                m  ...                          NaN            NaN   

      ('Tom', 0.0)  ('Snare', 0.688)  ('Snare', 0.375)  ('Snare', 0.125)  \
0              NaN               NaN               NaN               NaN   
1              NaN               NaN               NaN               NaN   
2              NaN               NaN               NaN               NaN   
3              NaN               NaN               NaN               NaN   
4              NaN               NaN               NaN               NaN   
...            ...               ...               ...               ...   
4828           NaN               NaN               NaN               NaN   
4829           NaN               NaN               NaN               NaN   
4830           NaN               NaN               NaN               NaN   
4831           NaN               NaN               NaN               NaN   
4832           NaN               NaN               NaN               NaN   

      ('Snare', 0.625)  ('Cencerro', 0.5)  ('Snare', 0.812)  \
0                  NaN                NaN               NaN   
1                  NaN                NaN               NaN   
2                  NaN                NaN               NaN   
3                  NaN                NaN               NaN   
4                  NaN                NaN          

In [34]:
midi_stats.iloc[:,0:100].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4833 entries, 0 to 4832
Data columns (total 100 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Unnamed: 0              4833 non-null   int64  
 1   first_time_signature    4833 non-null   object 
 2   cant_time_signatures    4833 non-null   int64  
 3   bpm                     4833 non-null   float64
 4   compases                4833 non-null   int64  
 5   cant_dist_chords        4833 non-null   int64  
 6   avg_notes_quantified    4833 non-null   float64
 7   tempo_changes           4833 non-null   int64  
 8   tonalidad               4833 non-null   object 
 9   tonalidad_escala        4833 non-null   object 
 10  scale_coverage          4833 non-null   float64
 11  mayor_chord_coverage    4833 non-null   float64
 12  minor_chord_coverage    4833 non-null   float64
 13  scale_note_avg_1        4829 non-null   float64
 14  scale_note_avg_2        4822 non-null  

In [36]:
midi_stats.iloc[:,100:200].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4833 entries, 0 to 4832
Data columns (total 100 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   length_note_1.25            2373 non-null   float64
 1   chord_1_3_6                 1222 non-null   float64
 2   chord_2_4_6                 865 non-null    float64
 3   avg_simult_Reed             919 non-null    float64
 4   inst_Bassoon                191 non-null    float64
 5   chord_2_4+_6                450 non-null    float64
 6   chord_2_4_5_7               550 non-null    float64
 7   avg_simult_Strings          1336 non-null   float64
 8   inst_Strings                464 non-null    float64
 9   inst_String ensamble        1144 non-null   float64
 10  inst_Oboe                   240 non-null    float64
 11  length_note_1.375           2132 non-null   float64
 12  ('Oboe', 0.25)              156 non-null    float64
 13  ('Oboe', 0.75)              173 

In [35]:
midi_stats.iloc[:,200:300].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4833 entries, 0 to 4832
Data columns (total 100 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   length_note_1.625         1680 non-null   float64
 1   inst_Trombone             207 non-null    float64
 2   ('Trombone', 0.0)         160 non-null    float64
 3   chord_2_4_7               237 non-null    float64
 4   avg_simult_Guitar         1981 non-null   float64
 5   inst_Steel Guitar         532 non-null    float64
 6   inst_Acoustic Bass        686 non-null    float64
 7   inst_Flute                413 non-null    float64
 8   ('Acoustic Bass', 0.375)  279 non-null    float64
 9   ('Acoustic Bass', 0.5)    612 non-null    float64
 10  ('Acoustic Bass', 0.625)  168 non-null    float64
 11  ('Acoustic Bass', 0.0)    654 non-null    float64
 12  ('Acoustic Bass', 0.25)   474 non-null    float64
 13  ('Acoustic Bass', 0.75)   545 non-null    float64
 14  ('Acous

In [37]:
midi_stats.iloc[:,300:400].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4833 entries, 0 to 4832
Data columns (total 100 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   ('Flute', 0.375)                  191 non-null    float64
 1   ('String ensamble', 0.969)        162 non-null    float64
 2   inst_Electric Piano               836 non-null    float64
 3   ('Electric Piano', 0.0)           795 non-null    float64
 4   ('Electric Piano', 0.125)         402 non-null    float64
 5   ('Electric Piano', 0.875)         465 non-null    float64
 6   ('Electric Piano', 0.625)         430 non-null    float64
 7   ('Electric Piano', 0.375)         468 non-null    float64
 8   ('Electric Piano', 0.75)          578 non-null    float64
 9   ('Electric Piano', 0.5)           700 non-null    float64
 10  ('Electric Piano', 0.25)          584 non-null    float64
 11  ('Electric Piano', 0.688)         151 non-null    float64
 12  ('Ele

In [6]:
midi_stats.describe().T[['mean','std','min','50%','75%','max']].head(50)

mean          std        min          50%  \
Unnamed: 0              2639.000000  1524.060366   0.000000  2639.000000   
cant_time_signatures       2.265581     8.029451   1.000000     1.000000   
bpm                      108.611858    39.531545   0.000000   107.000000   
compases                 112.298352   118.606443   5.000000    95.000000   
avg_notes_quantified       0.262145     0.319898   0.000000     0.089672   
tempo_changes              0.293427     0.455375   0.000000     0.000000   
scale_coverage            89.925942    11.752697   3.370000    93.350000   
mayor_chord_coverage      46.401826    12.631795   0.000000    45.980000   
minor_chord_coverage      43.400144    11.296210   0.000000    43.420000   
scale_note_avg_1          21.964689     8.606316   0.020000    20.535000   
scale_note_avg_2          13.946603     6.683328   0.010000    14.675000   
scale_note_avg_3          13.706269     4.906125   0.020000    13.955000   
scale_note_avg_4          11.927113     4.737184   0.020000    11.940000   
scale_note_avg_5          12.546246     5.580362   0.040000    11.480000   
scale_note_avg_6           8.428031     4.043447   0.010000     8.310000   
scale_note_avg_7           7.756880     4.154132   0.010000     7.210000   
avr_vertical_notes         0.870172     0.134447   0.000000     0.915278   
cant_eventos_por_pedal     5.296865    14.954135   0.000000     1.975490   
cant_pedales_seg                inf          NaN   0.014084     1.340863   
duracion_seg             227.561973   207.598515  13.125000   201.708984   
tracks_used                6.115173     4.117725   0.000000     5.000000   
avg_simult_Piano           0.815515     0.224948   0.000000     0.897843   
Piano                      6.675767     6.925636   0.002421     5.556493   
note_num_mean             58.726152     5.304804  34.001000    59.234801   
note_num_std              13.139935     4.654628   3.022767    12.278843   
note_num_min              24.885584    13.675715   0.000000    28.000000   
note_num_25%              51.309481     8.429887   0.000000    53.000000   
note_num_50%              60.867115     5.675638  27.000000    61.000000   
note_num_75%              67.836901     5.326580  35.000000    67.000000   
note_num_max              85.866452     8.861518  60.000000    85.000000   
Octave_mean                4.451246     0.434460   2.282000     4.491437   
Octave_std                 1.114072     0.341025   0.233785     1.059339   
Octave_min                 1.755825     1.024913   0.000000     2.000000   
Octave_25%                 3.795795     0.704959   0.000000     4.000000   
Octave_50%                 4.661205     0.572900   2.000000     5.000000   
Octave_75%                 5.204632     0.517414   2.000000     5.000000   
Octave_max                 6.741239     0.769410   5.000000     7.000000   
duration_mean            241.755103   263.222744  12.306107   147.414944   
duration_std             172.768379   197.839743   0.000000   108.579214   
duration_min              76.508392   130.684632   0.000000    30.000000   
duration_25%             125.321699   166.775101   0.000000    60.000000   
duration_50%             198.559292   266.236886   0.000000    96.000000   
duration_75%             289.173044   309.290755  12.000000   192.000000   
duration_max             691.983813   815.767320  24.000000   432.000000   
ff                         0.244228     0.300635   0.000000     0.105221   
pp                         0.072395     0.152723   0.000000     0.012226   
f                          0.174425     0.200425   0.000000     0.119091   
mf                         0.159016     0.176650   0.000000     0.111827   
m                          0.289077     0.353469   0.000000     0.140436   
p                          0.060860     0.114237   0.000000     0.009660   

                                75%           max  
Unnamed: 0              3958.500000  5.278000e+03  
cant_time_signatures       1.000000  3.450000e+02 

In [7]:
midi_stats.Genero.value_counts()             

ClassicRock     2428
Classical       1507
Hymns            322
Jazz             313
latina           254
NEW-AGE          166
Dance_Techno     152
RapMusic          58
REGGAE            43
TANGO             36
Name: Genero, dtype: int64

mask_1 = (midi_stats.avg_notes_quantified > 0.4)
mask_2 = (midi_stats.bpm == 120)
midi_stats[mask_1 and mask_2]

In [8]:
midi_stats.Grupo.value_counts().head(30)

Bach                          981
Beatles                       209
Beethoven                     174
Midi New Real Book I          159
Midi New Real Book II         154
Vivaldi                       102
Schubert                      101
ONBE~169                       96
Chopin                         93
Metallica                      74
LedZeppelin                    64
PIANO~43                       53
ONBEK~41                       38
Brahms                         37
Madonna                        37
BeachBoys                      32
ENYA                           29
ELO                            26
RollingStones                  24
Eagles                         23
BryanAdams                     21
Trance                         20
PaulMcCartney                  20
Nirvana                        19
BillyJoel                      19
Rachmaninov                    19
Genesis                        18
CreedenceClearwaterRevival     18
Carpennters                    17
Styx          

In [9]:
midi_stats.cant_time_signatures.value_counts()

1      4401
3       198
2       194
5        65
7        65
4        57
9        56
6        38
8        27
11       25
12       16
13       15
15       12
10       10
14       10
18        8
17        7
21        7
25        7
19        5
16        4
40        3
42        3
24        3
36        3
29        2
20        2
28        2
46        2
31        2
27        2
23        2
22        2
44        2
56        1
32        1
52        1
259       1
33        1
37        1
119       1
83        1
59        1
55        1
51        1
43        1
39        1
35        1
58        1
34        1
171       1
117       1
57        1
53        1
41        1
345       1
Name: cant_time_signatures, dtype: int64

In [10]:
len_mask = midi_stats.cant_time_signatures > 1
midi_stats[len_mask]

Unnamed: 0 first_time_signature  cant_time_signatures    bpm  compases  \
36            36                  4/4                     3  100.0        41   
37            37                  4/4                     3  100.0        46   
113          113                  4/4                     3   70.0        47   
148          148                  3/4                    19  104.0        18   
184          184                  4/4                     3  100.0        46   
...          ...                  ...                   ...    ...       ...   
5208        5208                  4/4                     2   80.0        78   
5212        5212                  4/4                     2   96.0        91   
5223        5223                  4/4                     3   98.0       104   
5235        5235                  4/4                     7   78.0        82   
5260        5260                  4/8                     2   64.0        64   

      avg_notes_quantified  tempo_changes tonalidad tonalidad_escala  \
36                0.000000              0         C                M   
37                0.000000              0         C                M   
113               0.000000              0         F                M   
148               0.000000              0         C                M   
184               0.000000              0         G                M   
...                    ...            ...       ...              ...   
5208              0.396411              0         G                M   
5212              0.332193              0         C                M   
5223              0.692849              0         D                m   
5235              1.000000              0         F                M   
5260              0.002966              1        D#                m   

      scale_coverage  ...  ('Distortion Guitar', 0.625)  \
36             93.24  ...                           NaN   
37             96.51  ...                           NaN   
113            96.62  ...                           NaN   
148            98.59  ...                           NaN   
184            93.13  ...                           NaN   
...              ...  ...                           ...   
5208           95.00  ...                           NaN   
5212           98.77  ...                           NaN   
5223           99.05  ...                           NaN   
5235          100.00  ...                           NaN   
5260           81.16  ...                           NaN   

      ('Distortion Guitar', 0.375)  ('Distortion Guitar', 0.75)  \
36                             NaN                          NaN   
37                             NaN                          NaN   
113                            NaN                          NaN   
148                            NaN                          NaN   
184                            NaN                          NaN   
...                            ...                          ...   
5208                           NaN                          NaN   
5212                           NaN                          NaN   
5223                           NaN                          NaN   
5235                           NaN                          NaN   
5260                           NaN                          NaN   

      ('Distortion Guitar', 0.25)  ('Tom', 0.25)  ('Snare', 0.688)  \
36                            NaN            NaN               NaN   
37                            NaN            NaN               NaN   
113                           NaN            NaN               NaN   
148                           NaN            NaN               NaN   
184                           NaN            NaN               NaN   
...                           ...            ...               ...   
5208                          NaN            NaN               NaN   
5212                          NaN            NaN               NaN   
5223                          NaN   

In [30]:
tema_analysis = midi_stats.loc[148,:]
tema_analysis.dropna(inplace=True)
tema_analysis.drop(['info_tracks'],inplace=True)

C:\Users\Patricio\anaconda3\envs\dhdsblend\lib\site-packages\pandas\core\series.py:4494: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._update_inplace(result)
C:\Users\Patricio\anaconda3\envs\dhdsblend\lib\site-packages\pandas\core\generic.py:3939: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._update_inplace(obj)


In [31]:
pd.DataFrame(tema_analysis.iloc[0:50])

148
Unnamed: 0                 148
first_time_signature       3/4
cant_time_signatures        19
bpm                        104
compases                    18
avg_notes_quantified         0
tempo_changes                0
tonalidad                    C
tonalidad_escala             M
scale_coverage           98.59
mayor_chord_coverage     70.36
minor_chord_coverage     49.99
scale_note_avg_1         37.96
scale_note_avg_2         24.07
scale_note_avg_3         11.11
scale_note_avg_4          8.33
scale_note_avg_5          6.94
scale_note_avg_6          6.48
scale_note_avg_7           3.7
avr_vertical_notes    0.944444
duracion_seg           32.8846
tracks_used                  3
avg_simult_Piano      0.944444
Piano                  6.56842
note_num_mean               49
note_num_std           26.8025
note_num_min                 0
note_num_25%             47.75
note_num_50%                60
note_num_75%                67
note_num_max                79
Octave_mean            3.77315
Octave_std              2.0886
Octave_min                   0
Octave_25%                3.75
Octave_50%                   5
Octave_75%                   5
Octave_max                   6
duration_mean          756.148
duration_std           458.458
duration_min                 0
duration_25%               512
duration_50%              1024
duration_75%              1024
duration_max              1536
ff                    0.787037
pp                    0.212963
f                            0
mf                           0
m                            0

In [32]:
pd.DataFrame(tema_analysis.iloc[50:100])

148
p                          0
0.5                 0.148148
1.0                 0.569444
0.0                 0.212963
0.25              0.00462963
Genero             Classical
Grupo                   Bach
tema                013705ch
('Piano', 0.0)       4.05556
('Piano', 0.333)     2.44444
('Piano', 0.667)     4.05556
('Piano', 0.5)       1.05556
('Piano', 0.833)    0.333333
0.75              0.00462963
1.5                0.0601852